src/lib/node/message_handler.ts

```diff
@@ -73,9 +79,14 @@ export class NodeMessageHandler {
           actualFirmware.data,
           actualFirmware.firmwareTarget
         );
+        this.firmwareUpdateQueued[nodeId] = true;
+        node.once("firmware update progress", (_, __, ___) => {
+          this.firmwareUpdateQueued[nodeId] = false;
+        });
         return {};
       case NodeCommand.abortFirmwareUpdate:
         await node.abortFirmwareUpdate();
+        this.firmwareUpdateQueued[nodeId] = false;
```

Comment:
What will happen to the event listener now?

Reply:
(this is not needed to clean up if we just attach the listener once per run of the server)

Reply:
good point